### PR TITLE
Adding new CRUD methods

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: bigml/api.py [hunk 1]
```diff
@@ -198,6 +198,19 @@ def count(listing):
...
 
+def filter_kwargs(kwargs, list_of_keys, out=False):
+    """Creates a new dict with the selected list of keys if present
+    If `out` is set to True, the keys in the list are removed
+    If `out` is set to False, only the keys in the list are kept
+
+    """
+    new_kwargs = {}
+    for key in kwargs:
+        if out:
```
Comment (jaor) [inline bigml/api.py:209]: this `if out:` is a leftover, no?

File: bigml/api.py [hunk 2]
```diff
@@ -362,6 +375,74 @@ def __init__(self, username=None, api_key=None,
...
+        """Create resources
+
+        """
+        finished = kwargs.get('finished', True)
+        create_kwargs = filter_kwargs(kwargs,
+                                      ['query_string', 'finished'],
+                                      out=True)
+        try:
+            resource_info = self.creaters[resource_type](*args,
+                                                         **create_kwargs)
```
Comment (jaor) [inline bigml/api.py:388]: i'd call them creators, creaters sounds weird

File: docs/index.rst [hunk 3]
```diff
@@ -414,6 +414,35 @@ You can also generate an evaluation for the model by using:
     evaluation = api.create_evaluation(model, test_dataset)
     api.ok(evaluation)
 
+
+The API object also offers some ``create``, ``get``, ``update`` and ``delete``
```
Comment (jaor) [inline docs/index.rst:418]: i'd say "the" rather than "some"

File: docs/index.rst [hunk 4]
```diff
@@ -414,6 +414,35 @@ You can also generate an evaluation for the model by using:
...
+.. code-block:: python
+
+    from bigml.api import BigML
+
+    api = BigML()
+
+    source = api.create('source', './data/iris.csv')
+    source = api.update(source, {"name": "my new source name"})
+
+Note that these methods don't need the ``api.ok`` to be called to wait for
```
Comment (jaor) [inline docs/index.rst:431]: "the api.ok() function/method"

File: docs/index.rst [hunk 5]
```diff
@@ -414,6 +414,35 @@ You can also generate an evaluation for the model by using:
...
+the resource to be finished. The method waits internally for it and by default.
```
Comment (jaor) [inline docs/index.rst:432]: and by default -> by default

File: docs/index.rst [hunk 6]
```diff
@@ -414,6 +414,35 @@ You can also generate an evaluation for the model by using:
...
+.. code-block:: python
+
+    from bigml.api import BigML
+
+    api = BigML()
+
+    source = api.create('source', './data/iris.csv')
+    dataset = api.create('dataset', source, finished=False) # unfinished
+    api.ok(dataset) # waiting explicitly for the dataset to finish
+
```
Comment (jaor) [inline docs/index.rst:445]: maybe we could briefly mention the other generic methods (get/delete take only
the resource id), and show an example of update with finished=False.

File: bigml/api.py (hunk 1 above)
Comment (mmerce) [reply bigml/api.py:209]: yes

File: bigml/api.py (hunk 2 above)
Comment (mmerce) [reply bigml/api.py:388]: oh, that's been there for some time...

**Review by jaor — APPROVED** [review]

**Pull request merged**